### PR TITLE
Relax join reorder limitation

### PR DESCRIPTION
**Pull request by liuyehcf**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Allow join reorder rule performed when lack of column state info

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/join/ReorderJoinRule.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 6 / 8 (75.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/Utils.java|5|7|71.43%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/join/ReorderJoinRule.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 6 / 8 (75.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/Utils.java|5|7|71.43%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/join/ReorderJoinRule.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 6 / 6 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/join/ReorderJoinRule.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/Utils.java|5|5|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Utils.java [hunk 1]
```diff
@@ -370,8 +372,16 @@ public static boolean hasUnknownColumnsStats(OptExpression root) {
                     scanOperator.getColRefToColumnMetaMap().values().stream().map(Column::getName).collect(
                             Collectors.toList());
             if (operator instanceof LogicalOlapScanOperator) {
+                Table table = scanOperator.getTable();
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Utils.java:375]: I think need to distinguish key columns and value columns for aggregate table.

Comment (liuyehcf) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Utils.java:375]: Seems hard to making precise judgments

**Review by Seaven — DISMISSED** [review]

**Comment by Seaven** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 6 / 6 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/join/ReorderJoinRule.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/Utils.java|5|5|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Utils.java (hunk 1 above)
Comment (kangkaisen) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Utils.java:375]: ColumnRefFactory could get Column， Column has `isKey` method

Comment (Seaven) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Utils.java:375]: can be check aggreagte key through ScanNode.columnMetaToColRefMap @liuyehcf

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 25 / 27 (92.59%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/Utils.java|16|18|88.89%|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/scalar/ScalarOperator.java|4|4|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/join/ReorderJoinRule.java|5|5|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Utils.java [hunk 2]
```diff
@@ -362,21 +364,37 @@ public static boolean capableSemiReorder(OptExpression root, boolean hasSemi, in
...
+                if (table instanceof OlapTable) {
+                    // Currently, aggregate model has no column statistical information
+                    // It is still allowed to perform ReorderJoinRule if aggregate key is not in the join predicates
+                    if (KeysType.AGG_KEYS.equals(((OlapTable) table).getKeysType())) {
+                        Optional<Column> joinPredicatesContainsKeyColumn = joinPredicateColumnRefs.stream()
+                                .filter(colRefToColumnMetaMap::containsKey)
+                                .map(colRefToColumnMetaMap::get)
+                                .filter(Column::isKey)
```
[the remark this answers is not in the document]
Comment (liuyehcf) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Utils.java]: updated

**Review by kangkaisen — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 9 / 12 (75.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/Utils.java|8|11|72.73%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/join/ReorderJoinRule.java|1|1|100.00%|

**Comment by liuyehcf** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 11 / 13 (84.62%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/Utils.java|10|12|83.33%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/join/ReorderJoinRule.java|1|1|100.00%|

**Review by Seaven — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**